src/mlagility/api/script_api.py

```diff
@@ -146,8 +148,27 @@ def benchmark_script(
     models_found: Dict[str, ModelInfo] = {}
 
     for script in input_scripts:
+        script_path, targets, encoded_input = decode_input_script(script)
+
+        # Filter scripts by label
+        if labels != []:
+            skip_script = False
+            user_labels = lab.to_dict(labels)
+            file_labels = lab.load_from_file(encoded_input)
+            for key in user_labels:
+                # Skip benchmarking if user_label is not in file_labels
+                if key not in file_labels:
+                    skip_script = True
+                    break
+                # A label key may point to multiple label values
+                # Skip if not all values of user_labels[key] are in file_labels[key]
+                elif not all(elem in user_labels[key] for elem in file_labels[key]):
```

Comment:
Isn't this requirement a superset of the `if key not in file_labels:` requirement? (can't you just delete `if key not in file_labels:`?)

Reply:
Or maybe I'm just confused... 

This statement is using `user_labels[key]` but it's in a loop that's doing `for key in user_labels:`


that could at least be `for key, value in user_labels.items()` and then `all(elem in value...` right?

Reply:
I think you are a bit confused.

`[elem in required_labels[key] for elem in script_labels[key]]` is inside a loop that is iterating over all keys of `required_labels`. If a key of `required_labels` doesn't exist in `script_labels` that will result in a `KeyError` exception when evaluating `script_labels[key]`. 

This is why checking `if key not in script_labels` is needed.

I slightly modified the comments to clarify that.

Reply:
FYI, you can use `script_labels.get(key)` if you want to avoid `KeyError`.